feat(fromMethods): add fromDocuments and fromTexts

Itzaprado opened the pull request:
[ ] Add fromTexts and fromDocuments methods
[ ] Add tests for fromMethods
[ ] Fix test that marked error in the editor

Itzaprado committed:
feat(fromMethods): add fromDocuments and fromTexts
[ ] Add fromTexts and fromDocuments methods
[ ] Add tests for fromMethods
[ ] Fix test that marked error in the editor

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/vectorStore.ts:
```diff
@@ -179,6 +185,39 @@ class PostgresVectorStore extends VectorStore {
     )
   }
 
+  static async fromTexts(_texts: string[], _metadatas: object[], _embeddings: Embeddings, _dbConfig: dbConfigArgs): Promise<VectorStore> {
```
I prefer to use variables without the "_" prefix. 
I believe we should be using the more generic interface of EmbeddingsInnterface instead of the class Embeddings to ensure more flexibility.

averikitsch replied:
We also need metadata to support object[] | object. When constructing the Document we can jus add `const metadata = Array.isArray(metadatas) ? metadatas[i] : metadatas;`

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/vectorStore.ts:
```diff
@@ -179,6 +185,39 @@ class PostgresVectorStore extends VectorStore {
     )
   }
 
+  static async fromTexts(_texts: string[], _metadatas: object[], _embeddings: Embeddings, _dbConfig: dbConfigArgs): Promise<VectorStore> {
```
Do we need the interface or can we set this to be:
```js
dbConfig: {
  engine: PostgresEngine;
  tableName: string;
  config?: PostgresVectorStoreArgs;
}

Itzaprado replied:
Could you please clarify what you mean? I added something similar on line 25, and that's what I used to get the values ​​and pass them to the create method.

averikitsch replied:
Sorry, this meant to say do we need to define a new interface? You are right, were we should just have the interface defined.

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/vectorStore.ts:
```diff
@@ -179,6 +185,39 @@ class PostgresVectorStore extends VectorStore {
   }
 
+  static async fromTexts(_texts: string[], _metadatas: object[], _embeddings: Embeddings, _dbConfig: dbConfigArgs): Promise<VectorStore> {
+    const engine = _dbConfig.engine;
+    const tableName = _dbConfig.tableName;
```
Do we need to check if engine and tableName is not undefined?

Itzaprado replied:
I don't think so, they are required arguments and are defined as such in the dbConfigArgs interface.

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/vectorStore.ts:
```diff
@@ -179,6 +185,39 @@ class PostgresVectorStore extends VectorStore {
+      documents.push(doc);
+    } 
+
+    await vectorStore.addVectors(vectors, documents)
```
remove vector store and vector creation above and just use `return PostgresVectorStore.fromDocuments(...)`

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/vectorStore.ts:
```diff
@@ -179,6 +185,39 @@ class PostgresVectorStore extends VectorStore {
+    return vectorStore;
+  }
+
+  static async fromDocuments(_docs: Document<Record<string, any>>[], _embeddings: Embeddings, _dbConfig: dbConfigArgs): Promise<VectorStore> {
```
See comments from above and simplify docs to Document[]

Itzaprado committed:
refactor: update fromTexts and fromDocuments
[ ] remove _ in all variables
[ ] update fromTexts logic
[ ] update metadatas assigment

averikitsch commented on libs/langchain-google-cloud-sql-pg/src/vectorStore.ts:
```diff
@@ -185,33 +185,25 @@ class PostgresVectorStore extends VectorStore {
-    const tableName = _dbConfig.tableName;
-    const dbConfig = _dbConfig.dbConfig;
-    const vectorStore = await this.create(engine, _embeddings, tableName, dbConfig);
+  static async fromDocuments(_docs: Document[], embeddings: EmbeddingsInterface, dbConfig: dbConfigArgs): Promise<VectorStore> {
```
Remove underscore from _docs varialbe

Itzaprado committed:
refactor: remove unnecessary _ from docs
